### PR TITLE
Support for typescript

### PR DESCRIPTION
**Pull request by ddzy**
**Add a type definition for typescript, update README**

### CODE REVIEW

**Comment by whxaxes** [comment]
Hi, plz add an unittest for types

File: package.json [hunk 1]
```diff
@@ -16,13 +16,15 @@
     "test-optimized": "node --allow-natives-syntax --trace_opt --trace_deopt test/optimized.js"
   },
   "dependencies": {
+    "@types/node": "^10.12.12",
```
Comment (whxaxes) [inline package.json]: types should move to devDeps

**Comment by ddzy** [comment]
> Hi, plz add an unittest for types

😊😊haode, ok

File: package.json (hunk 1 above)
Comment (ddzy) [reply package.json]: > types should move to devDeps

It's my first time to make PR, i think it will do better.😂😂

Comment (whxaxes) [reply package.json]: It doesn't make sense for adding `@types/node` in `dependencies`. `dependencies` are required to run.

**Pull request closed without merging**